### DISCUSSION TITLE
Issue with Retrieving YouTube Transcripts for Non-English Languages

### DISCUSSION THREAD

**Discussion by VladimirMonin** [logancyang/obsidian-copilot#883] [Ideas]
I am currently testing the Plus version, and I must say, I am very pleased; it’s simply outstanding work!

However, I have encountered an issue while trying to retrieve transcripts for YouTube videos. Specifically, when I attempt to get the transcript for the video at the link https://www.youtube.com/watch?v=wZXk8y4fEYA, the program returns an error. The error message states that it could not retrieve a transcript for this video and indicates that no transcripts were found for the requested language codes, which in this case is English.

Upon reviewing the logs, I noticed that the program mentions available transcripts only in Russian, both as manually created and auto-generated options. This situation raises a concern that there may be a broader issue with the API's ability to handle languages other than English. If transcripts are not retrievable for English, it is likely that users trying to access transcripts in other languages might face similar problems.

I would appreciate it if you could investigate this matter further. It seems that there may be a need for improved handling of transcripts across various languages in the API. Thank you for your incredible work!

**Comment by logancyang** [comment]
@VladimirMonin thanks for testing plus alpha! The youtube tool currently only supports English videos (with auto-transcription option on, it's a youtube setting at upload time), I mentioned it in the demo: https://youtu.be/tZ2jm_UPc6c?si=4ubRX6Y4-4-XBkIR&t=725

Other languages will come later. Stay tuned!
